packages/widgetbook/lib/src/knobs/nullable_bool_knob.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import 'package:widgetbook/src/knobs/knobs.dart';
+import 'package:widgetbook/src/knobs/nullable_checkbox.dart';
+
+class NullableBoolKnob extends Knob<bool?> {
+  NullableBoolKnob({
+    required String label,
+    String? description,
+    required bool? value,
+  }) : super(
+          label: label,
+          description: description,
+          value: value,
+        );
+
+  @override
+  Widget build() => NullableBooleanKnobWidget(
+        label: label,
+        description: description,
+        value: value,
+      );
+}
+
+class NullableBooleanKnobWidget extends StatefulWidget {
+  const NullableBooleanKnobWidget({
+    Key? key,
+    required this.label,
+    required this.description,
+    required this.value,
+  }) : super(key: key);
+
+  final String label;
+  final String? description;
+  final bool? value;
+
+  @override
+  State<NullableBooleanKnobWidget> createState() =>
+      _NullableBooleanKnobWidgetState();
+}
+
+class _NullableBooleanKnobWidgetState extends State<NullableBooleanKnobWidget> {
+  bool value = false;
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget.value != null) {
+      value = widget.value!;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final _disabled = widget.value == null;
+    return ListTile(
+      title: Text(widget.label, overflow: TextOverflow.ellipsis),
+      dense: true,
+      subtitle: widget.description == null ? null : Text(widget.description!),
+      trailing: Wrap(
+        crossAxisAlignment: WrapCrossAlignment.center,
+        children: [
+          Switch(
+            key: Key('${widget.label}-switchTileKnob'),
+            value: value,
+            onChanged: _disabled
+                ? null
+                : (v) {
+                    setState(() {
+                      value = v;
+                    });
+                    context.read<KnobsNotifier>().update(widget.label, v);
+                  },
+          ),
+          NullableCheckbox(
+            cachedValue: value,
+            value: widget.value,
+            label: widget.label,
+          ),
+        ],
+      ),
+    );
+  }
+}
```

Comment:
while this seems to work, it requires a lot of code to recover the value whenever the null values has been selected. Is it possible to track the non-null value within the provider and recover it once the null value is not longer selected?

Reply:
Just realized that it might be hard to actually save the value for multiple know because the types of value across multiple knobs might not be the same. Therefore, it might be hard to save this in some sort of `List` or `Map`.. Let me know what you think 😀

Reply:
I didn't think about saving the previous value inside the provider, but I see how it would possibly work. If we were to do the map we'd have to do type casting everytime we got the previous value, unless there's some kind of way with generics to say what kind of type the value is based on the key (I'm not sure if there's a way like this). I implemented a few widgets like this and it wasn't too difficult, but let me know if you think of another way that may be better

Reply:
due to type casting it might be difficult to implement this in the provider because we also want users to be able to expand custom knobs in case we are not supporting their case. Therefore, keeping the state in the Widget is fine for now.